editor/src/components/canvas/canvas-strategies/flex-align-parent-strategy.ts

```diff
@@ -0,0 +1,260 @@
+import { foldEither } from '../../../core/shared/either'
+import * as EP from '../../../core/shared/element-path'
+import {
+  ElementInstanceMetadata,
+  emptyComments,
+  jsxAttributeValue,
+  JSXElement,
+} from '../../../core/shared/element-template'
+import { setJSXValuesAtPaths, ValueAtPath } from '../../../core/shared/jsx-attributes'
+import {
+  canvasPoint,
+  CanvasPoint,
+  CanvasRectangle,
+  magnitude,
+  rectContainsPoint,
+} from '../../../core/shared/math-utils'
+import { ElementPath } from '../../../core/shared/project-file-types'
+import * as PP from '../../../core/shared/property-path'
+import {
+  EditorState,
+  forUnderlyingTargetFromEditorState,
+  modifyUnderlyingForOpenFile,
+  TransientFilesState,
+} from '../../editor/store/editor-state'
+import {
+  CanvasStrategy,
+  CanvasStrategyUpdateFnResult,
+  FlexAlignControlRectProps,
+  SelectModeCanvasSessionProps,
+  SelectModeCanvasSessionState,
+} from './canvas-strategy-types'
+
+export const flexAlignParentStrategy: CanvasStrategy = {
+  name: "Change Parent's Flex Align and Justify",
+  fitnessFn: (editor, currentSession) => {
+    if (editor.selectedViews.length === 1) {
+      const selectedView = editor.selectedViews[0]
+
+      const isFlexLayouted = MetadataUtils.isParentYogaLayoutedContainerForElementAndElementParticipatesInLayout(
+        MetadataUtils.findElementByElementPath(editor.jsxMetadata, selectedView),
+      )
+      const hasNoSiblings = MetadataUtils.getSiblings(editor.jsxMetadata, selectedView).length === 1
+
+      if (isFlexLayouted && hasNoSiblings) {
+        return 10 // fit!
+      }
+    }
+    return 0 // not fit
+  },
+  updateFn: (
+    lifecycle: 'transient' | 'final',
+    editorState: EditorState,
+    sessionProps: SelectModeCanvasSessionProps,
+    sessionState: SelectModeCanvasSessionState,
+  ): CanvasStrategyUpdateFnResult => {
+    // only apply after a certain treshold IF we hadn't already passed that treshold once
+    const draggedElement = editorState.selectedViews[0]
+
+    if (
+      !sessionState.dragDeltaMinimumPassed &&
+      magnitude(sessionProps.drag ?? canvasPoint({ x: 0, y: 0 })) < 15
+    ) {
+      return {
+        newSessionState: sessionState,
+        transientFilesState: {},
+        editorStatePatch: {
+          canvas: {
+            controls: {
+              animatedPlaceholderTargetUids: {
+                $set: lifecycle === 'transient' ? [EP.toUid(draggedElement)] : [],
+              },
+            },
+          },
+        },
+      }
+    }
+
+    const targetParent = MetadataUtils.getParent(editorState.jsxMetadata, draggedElement)
+    const indicatorBoxes = calcualteFlexAlignIndicatorBoxes(
+      targetParent,
+      sessionProps.mousePosition,
+    )
+
+    // if any indicator box is highlighted, we also want to change the parent's style too
+    const higlightedIndicator = indicatorBoxes.filter((b) => b.highlighted === true)[0]
```

Comment:
`find` just gives us the first one anyway right?
```suggestion
    const higlightedIndicator = indicatorBoxes.find((b) => b.highlighted === true)
```